girder_tech_journal/__init__.py

```diff
@@ -36,13 +36,52 @@
 from girder.models.user import User
 from girder.utility import mail_utils
 from girder.utility.model_importer import ModelImporter
+from girder.plugin import GirderPlugin
 from girder import events
 from girder_worker_utils.transforms.girder_io import GirderUploadToItem
 from tech_journal_tasks.tasks import processGithub, surveySubmission
 from . import constants
 from .models.journal import download_statistics, Journal
 
 
+class TechJournalPlugin(GirderPlugin):
+    DISPLAY_NAME = 'Tech Journal'
+    CLIENT_SOURCE_PATH = 'web_client'
+
+    def load(self, info):
+        techJournal = TechJournal()
+        techJournal.download_statistics = download_statistics()
+        techJournal.journal_collection = Journal()
+        ModelImporter.registerModel('journal', Journal, plugin='tech_journal')
+        mail_utils.addTemplateDirectory(os.path.join(
+            os.path.dirname(__file__),
+            'mail_templates'
+        ))
+
+        events.bind('model.setting.validate', 'journalMain', validateSettings)
+        info['apiRoot'].journal = techJournal
```

Comment:
Let's put this assignment onto `info['apiRoot']` next to the `techJournal = TechJournal()`. Ideally, they could be combined into one line.